collaboration-service/index.js

```diff
@@ -0,0 +1,103 @@
+import express from "express";
+import cors from "cors";
+import { createServer } from "http";
+import mongoose from "mongoose";
+import "dotenv/config";
+import { Server, Socket } from "socket.io";
+import Room from "./model/room-model.js";
+import collaborationRoutes from "./routes/collaboration-routes.js";
+
+const uri =
+  process.env.ENV == "PROD"
+    ? process.env.DB_CLOUD_URI
+    : process.env.DB_LOCAL_URI;
+
+mongoose
+  .connect(uri)
+  .then((x) =>
+    console.log(
+      `Connected to MongoDB! Database name: "${x.connections[0].name}"`
+    )
+  )
+  .catch((err) => console.error("Error connecting to MongoDB", err.reason));
+
+const app = express();
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+app.use(cors());
+app.options("*", cors());
+
+// Routes tested via Postman
+app.use("/collab", collaborationRoutes);
+
+const httpServer = createServer(app);
+const PORT = process.env.PORT || 8003;
+
+httpServer.listen(PORT, () => {
+  console.log(`Listening on port ${PORT}`);
+});
+
+// Sockets tested using https://socketio.bloggernepal.com/4.0.1/
+// Also tested via Postman
+const io = new Server(httpServer, {
+  cors: {
+    origin: true,
+  },
+});
+
+// When a client initiate socket connection.
+io.on("connection", (socket) => {
+  console.log("A user connected");
+
+  socket.on("disconnect", () => {
+    console.log("a user disconnected");
+  });
+
+  // Client joins a specific room.
+  socket.on("CONNECTED", async ({ roomId, username }) => {
+    console.log(`Connecting to room ${roomId}`);
+    const { users } = await Room.findOne({ roomId: roomId });
+    console.log(users);
+    let newUsers = users;
+    newUsers.push(username);
+    await Room.findOneAndUpdate(
+      { roomId: roomId },
+      {
+        users: newUsers,
+      }
```

Comment:
Minor nitpick but good to standardize styling to one line as per above line.

Reply:
Done